src/request/mod.rs

```diff
@@ -1,105 +1,42 @@
-//! Module containing all structs modelling requests to the boomlings APIs.
+//! Module containing all structs modelling requests to the Boomlings APIs.
 //!
 //! These directly implement (de)serialization into RobTop's data format, unlike models where
 //! RobTop's eccentricities are hidden. This is since directly re-using these structs outside of
 //! making/proxying requests for the boomlings servers seems rather useless to me, as they already
 //! contain a lot of boomlings-specific fields.
+//! This can also be edited for a specific GDPS, e.g 1.9 GDPS. (hi absowute :3)
 
 use crate::{model::GameVersion, serde::RequestSerializer};
 use serde::{Deserialize, Serialize};
+use once_cell::sync::OnceCell;
 
-macro_rules! const_setter {
-    ($name: ident, $field: ident, $t: ty) => {
-        pub const fn $name(mut self, $field: $t) -> Self {
-            self.$field = $field;
-            self
-        }
-    };
-
-    ($name: ident, $t: ty) => {
-        pub const fn $name(mut self, arg0: $t) -> Self {
-            self.$name = arg0;
-            self
-        }
-    };
+static GDPS_URL: OnceCell<String> = OnceCell::new();
```

Comment:
this needs to be a `pub` static, so that code using dash-rs can set the `OnceCell` in their initialization code.

Also, let's name it `REQUEST_BASE`, because it doesn't have to be a GDPS URL (and in the most common usecase of interacting with the boomlings servers, it actually won't be a GDPS URL).

I also realized that this needs to be a `OnceLock` and not a `OnceCell` for use in statics. Sorry! So altogether, this should be

```suggestion
static REQUEST_BASE: OnceLock<&'static str> = OnceLock::new();
```